tests/granulate_utils/test_jvm_flags.py

```diff
@@ -0,0 +1,131 @@
+                JvmFlag(
+                    name="OptimizeFill",
+                    type="bool",
+                    value="true",
+                    origin="command line, ergonomic",
+                    kind=["C2", "product"],
+                ),
+                JvmFlag(name="OptimizePtrCompare", type="bool", value="true", origin="default", kind=["C2", "product"]),
+            ],
+        ),
+        (
+            """
+    uintx NewSize                                  := 357564416                           {product}
+    uintx NewSizeThreadIncrease                     = 5320                                {pd product}
+     intx NmethodSweepActivity                      = 10                                  {product}
+    uintx OldPLABWeight                             = 50                                  {lp64_product}
+    uintx OldSize                                  := 716177408                           {product}
+     bool OmitStackTraceInFastThrow                 = true                                {product}
+ccstrlist OnError                                  := cat hs_err_pid%p.log
+      OnError                             += ps -ef                              {product}
+ccstrlist OnOutOfMemoryError                        =                                     {product}
+     intx OnStackReplacePercentage                  = 140                                 {pd product}
+     bool OptoBundling                              = false                               {C2 pd product}
+""",
+            [
+                JvmFlag(name="NewSize", type="uintx", value="357564416", origin="non-default", kind=["product"]),
+                JvmFlag(
+                    name="NewSizeThreadIncrease", type="uintx", value="5320", origin="default", kind=["pd", "product"]
+                ),
+                JvmFlag(name="NmethodSweepActivity", type="intx", value="10", origin="default", kind=["product"]),
+                JvmFlag(name="OldPLABWeight", type="uintx", value="50", origin="default", kind=["lp64_product"]),
+                JvmFlag(name="OldSize", type="uintx", value="716177408", origin="non-default", kind=["product"]),
+                JvmFlag(
+                    name="OmitStackTraceInFastThrow", type="bool", value="true", origin="default", kind=["product"]
+                ),
+                JvmFlag(
+                    name="OnStackReplacePercentage", type="intx", value="140", origin="default", kind=["pd", "product"]
+                ),
+                JvmFlag(
+                    name="OptoBundling", type="bool", value="false", origin="default", kind=["C2", "pd", "product"]
+                ),
+            ],
+        ),
+    ],
+)
+def test_parse_jvm_flags(jvm_flags_string: str, expected_jvm_flags_list: List[JvmFlag]) -> None:
+    assert parse_jvm_flags(jvm_flags_string) == expected_jvm_flags_list
+
+
+@pytest.mark.parametrize(
+    "jvm_flag,expected_jvm_flag_serialized",
+    [
+        (
+            JvmFlag(
```

Comment:
Add something here that proves the sorting during serialization? (or parsing)

Reply:
i.e a test case

Reply:
serialization/deserialization is for a single flag, and parsing is not guaranteeing alphabetical sort, but a sorted list according to the input string (which is tested in the sanity case).

fixed the parsing to return a sorted list, as we were accounting on ordered `VM.flags` output string.

Reply:
The sorting is necessary when sending the flags as metadata, i.e., it should appear in the profiler code after filtering out the undesired flags. This module in granulate-utils only parses the output of `jcmd VM.flags` - it isn't under its responsibility to change their order.

Reply:
You are calling `sorted` in this code, so my suggestion is to incorporate a test that ensures sorting remains in effect.

Reply:
I called it because of the test

Reply:
k